src/Pages/SubBreedPage.tsx

```diff
@@ -0,0 +1,73 @@
+import axios from "axios";
+import { FunctionComponent, useEffect, useState } from "react";
+import ReactPaginate from "react-paginate";
+import { useParams } from "react-router-dom";
+import Dog from "../Component/Dog";
+//import * as helpers from "../Helpers";
+
+
+interface SubBreedPageProps {
+    
+}
+ 
+const SubBreedPage: FunctionComponent<SubBreedPageProps> = () => {
+    const POSTSPERPAGE = 10;
+    let {subBreed,breedName} = useParams<string>();
+    let [subBreedImgsContainer, setSubBreedImgsContainer] = useState<string[]>([]);
+    let [numOfPages,setNumOfPages] = useState<number>(0)
+    let [imgsInScreen,setImgsInScreen] = useState<string[]>([]);
+    const [compIsMounted,setCompIsMounted] = useState<boolean>(false); // Mounts once the fetchDogs function
+    const breedPageChange = ({selected}:any) => {
+        setImgsInScreen(subBreedImgsContainer.slice(selected*POSTSPERPAGE, selected*POSTSPERPAGE+POSTSPERPAGE))
+    }
+    useEffect( () => {
+        axios
+        .get('https://dog.ceo/api/breed/'+breedName+'/'+subBreed+'/images')
+        .then((imgs) => {
+            setSubBreedImgsContainer(imgs.data.message);
+            setNumOfPages(Math.ceil( imgs.data.message.length / POSTSPERPAGE ));
+            if (!compIsMounted){

```

Comment:
What is the exact usage of this variable?

Reply:
compIsMounted in a stage variable that I used to inside useEffect.
The problem it solves is this:

a) I wrote a function that fetches Dogs,
b) I called it inside useEffect Hook, as a dependency.
c) useEffect consistently executes this function because this function is called consistently.
d) With compIsMounted if the Component is not mounted, the fetchDogs function is called only once and the second time due to the if statement the fetchDogs function does not run again. ==> useEffect stops running.

Without this, the function fetchDogs runs continuously inside useEffect severely affecting performance.